Source/WebCore/html/canvas/CanvasRenderingContext2D.cpp

```diff
@@ -157,6 +157,22 @@ void CanvasRenderingContext2D::setFontWithoutUpdatingStyle(const String& newFont
     ASSERT(state().font.isPopulated());
 }
 
+std::optional<FilterOperations> CanvasRenderingContext2D::setFilterStringWithoutUpdatingStyle(const String& filterString)
+{
+    auto& document = canvas().document();
+    if (!document.settings().canvasFiltersEnabled())
+        return std::nullopt;
+
+    document.updateStyleIfNeeded();
```

Comment:
The "WithoutUpdatingStyle" in the name is confusing given that you update style here.

Reply:
The only change we do in the style is in `colorFromPrimitiveValueWithResolvedCurrentColor()` where we fix a few things for currentColor case. I changed the name of this function to be` createFilterOperations()`.